src/System.Text.Json/experiments/Json.Net.Test/Json.Net.Tests/JsonNetComparison.cs

```diff
@@ -0,0 +1,229 @@
+            }
+
+            Output("System.Text.Json Timing Results");
+            foreach (var res in TimingResultsJsonReader)
+            {
+                Output(res.ToString());
+            }
+            
+            TimingResultsJsonNet.Clear();
+            TimingResultsJsonReader.Clear();
+        }
+
+        private static void RunTestJsonNet(string str, bool output)
+        {
+            var utf8Str = new Utf8String(str);
+            Timer.Restart();
+            for (var i = 0; i < NumberOfIterations; i++)
+            {
+                JsonNetReaderHelper(new StringReader(utf8Str.ToString()), output);
+            }
+            TimingResultsJsonNet.Add(Timer.ElapsedMilliseconds);
+        }
+
+        private static void JsonNetReaderHelper(TextReader str, bool output)
+        {
+            var reader = new JsonTextReader(str);
+
+            while (reader.Read())
+            {
+                if (reader.Value != null)
+                {
+                    // ReSharper disable once UnusedVariable
+                    var x = reader.TokenType;
+                    // ReSharper disable once UnusedVariable
+                    var y = reader.Value;
+                    if (output) Console.WriteLine(y);
+                }
+                else
+                {
+                    // ReSharper disable once UnusedVariable
+                    var z = reader.TokenType;
+                    if (output) Console.WriteLine(z);
+                }
+            }
+        }
+
+        private static void RunTestJsonReader(string str, bool output)
+        {
+            var utf8Str = new Utf8String(str);
+            Timer.Restart();
+            for (var i = 0; i < NumberOfIterations; i++)
+            {
+                JsonReaderHelper(utf8Str, output);
+            }
+            TimingResultsJsonReader.Add(Timer.ElapsedMilliseconds);
+        }
+        
+        private static void JsonReaderHelper(Utf8String str, bool output)
+        {
+            var reader = new JsonReader(str);
+            while (reader.Read())
+            {
+                var tokenType = reader.TokenType;
+                switch (tokenType)
+                {
+                    case JsonReader.JsonTokenType.ObjectStart:
+                    case JsonReader.JsonTokenType.ObjectEnd:
+                    case JsonReader.JsonTokenType.ArrayStart:
+                    case JsonReader.JsonTokenType.ArrayEnd:
+                        if (output) Console.WriteLine(tokenType);
+                        break;
+                    case JsonReader.JsonTokenType.Property:
+                        var name = reader.GetName();
+                        if (output) Console.WriteLine(name);
+                        var x = reader.GetValue();
+                        if (output) Console.WriteLine(x);
+                        break;
+                    case JsonReader.JsonTokenType.Value:
+                        x = reader.GetValue();
```

Comment:
It returns an object but that object has already been converted into an end user value, e.g. string, int, long, DateTime, etc, that is set in the JObject DOM or the deserializing object. Casting != converting.